developer/src/kmcomp/kmcomp.dpr

```diff
@@ -29,7 +29,7 @@ uses
   utilstr in '..\..\..\common\windows\delphi\general\utilstr.pas',
   utilsystem in '..\..\..\common\windows\delphi\general\utilsystem.pas',
   utildir in '..\..\..\common\windows\delphi\general\utildir.pas',
-  utilkeyboard in '..\..\..\windows\src\global\delphi\general\utilkeyboard.pas',
+  utilkeyboard in '..\..\..\common\windows\delphi\keyboards\utilkeyboard.pas',
```

Comment:
Man, it'd be nice if we could just use `$KEYMAN_ROOT` in the include paths, wouldn't it?

Or if we could set an extra "include" root folder like I believe is possible in Visual Studio so that we didn't need the constant `../../..common/windows` stuff all over the place.

Reply:
It is possible to set an include folder. The problem with Delphi is it is too easy to drag in additional units which can and do cause trouble (particularly because units can be self-initializing). In a concrete example, we had command line projects which accidentally pulled in graphical ui (aka Visual Component Library, VCL) units, and that caused dialog boxes to pop on errors rather than having them reported via console messages...

So, long ago, we opted to be specific about unit usage, despite the pain in the .dpr. I would do more, but we are trying not to do new Delphi work.